src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -390,11 +385,9 @@ void InferRequest::SetGraph(std::shared_ptr<Graph> graph) {
         IE_THROW(NetworkNotLoaded);
     }
 
-    if (!m_graph->GetNetwork()->is_dynamic()) {
-        allocate_inputs();
-        allocate_outputs();
-        variables_states_ = m_graph->AllocateVariablesMemories();
-    }
+    allocate_inputs();
+    allocate_outputs();
```

Comment:
When it is dynamic, it tries to access network->output_memory in allocate_output(),  though it is not set yet. Is this handled in somewhere?

Reply:
Ah  I just checked it is no longer calling output memory but layout.